### PR TITLE
[coinbasepro] reconnectFailure & connectionSuccess NPE

### PR DESCRIPTION
**Pull request by mdvx**
return error for NPE instead of throwing one

### CODE REVIEW

File: xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java [hunk 1]
```diff
@@ -84,11 +84,17 @@ public Completable disconnect() {
 
   @Override
   public Observable<Throwable> reconnectFailure() {
+    if (streamingService == null)
+      return Observable.error(new NullPointerException("streamingService is null"));
```
Comment (badgerwithagun) [inline xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java]: How about an `IllegalStateException("Not connected")`? More descriptive.

**Review by badgerwithagun — CHANGES_REQUESTED** [review]

File: xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java (hunk 1 above)
Comment (mdvx) [reply xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java]: works for me too

File: xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java [hunk 2]
```diff
@@ -84,11 +84,17 @@ public Completable disconnect() {
 
   @Override
   public Observable<Throwable> reconnectFailure() {
+    if (streamingService == null)
```
Comment (badgerwithagun) [inline xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java:87]: Sorry to mess you around, but I've just had another think about this. I think it's perfectly reasonable to expect these observables to fire on initial connection, not just on automated reconnections.  It seems wrong that we can't register to hear about connections/reconnections until _after_ we've connected. Could we just return publishers here and hook them up to the `StreamingService` before connecting?

**Comment by mdvx** [comment]
replaced by https://github.com/bitrich-info/xchange-stream/pull/566

**Pull request closed without merging**